### PR TITLE
Update import attributes `assert` tests to current semantics

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
This was part of https://github.com/tc39/test262/pull/3843, but:
- I rewrote it from scratch to make sure that it was correct
- It just updates `assert` semantics, it does not add `with` tests yet (I'll do that in a follow-up)

There is some complexity related to tests with invalid attributes, that should throw when linking, so I have a bunch of tests to verify valid _syntax_ that will throw then during resolution.

### CODE REVIEW

File: test/language/expressions/dynamic-import/import-assertions/2nd-param-assert-enumeration-enumerable.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+import('./2nd-param_FIXTURE.json', options)
+  .then(function(module) {
+    assert.sameValue(module.default, 262);
+  })
+  .then($DONE, $DONE);
+
+assert.sameValue(log.length, 1);
+assert.sameValue(log[0], "type")
```
Comment (ljharb) [inline test/language/expressions/dynamic-import/import-assertions/2nd-param-assert-enumeration-enumerable.js:60]:
```suggestion
assert.compareArray(log, ['type']);
```

File: test/language/expressions/dynamic-import/import-assertions/2nd-param-assert-enumeration-enumerable.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        a. If Type(options) is not Object,
+           [...]
+        b. Let assertionsObj be Get(options, "assert").
+        c. IfAbruptRejectPromise(assertionsObj, promiseCapability).
+        d. If assertionsObj is not undefined,
+           i. If Type(assertionsObj) is not Object,
+              [...]
+           ii. Let keys be EnumerableOwnPropertyNames(assertionsObj, key).
+    [...]
+features: [dynamic-import, import-assertions, json-modules, Symbol, Proxy]
```
Comment (ljharb) [inline test/language/expressions/dynamic-import/import-assertions/2nd-param-assert-enumeration-enumerable.js:25]:
```suggestion
features: [dynamic-import, import-assertions, json-modules, Symbol, Proxy]
includes:
  - compareArray.js
```

**Review by ljharb — APPROVED** [review]
LGTM but i'd prefer more maintainers to approve before merging

File: test/language/expressions/dynamic-import/import-assertions/2nd-param-assert-enumeration-enumerable.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+includes:
+  - compareArray.js
```
Comment (nicolo-ribaudo) [inline test/language/expressions/dynamic-import/import-assertions/2nd-param-assert-enumeration-enumerable.js:27]:
```suggestion
includes: [compareArray.js]
```

**Pull request merged**